### ISSUE TITLE
This line breaks syntax coloring when editing commit messages

### ISSUE THREAD

**Issue by gonzus** [nvim-lua/kickstart.nvim#237]
https://github.com/nvim-lua/kickstart.nvim/blob/4a37a0a9b1d53fd355637256ac975c619226fb6d/init.lua#L296

With this line, when I change one of my repos and then issue a `git commit -v` command (with `$EDITOR` set to neovim), the commit message itself has correct syntax coloring, but the diffs shown below the commit message show up as plain gray. Commenting the line out fixes the symptom.

**Comment by kang8** [comment]
Not reproduced on my end. 

Are you using the latest master branch? Or did you add a custom configuration?

**Comment by gonzus** [comment]
I just recloned from `git@github.com:nvim-lua/kickstart.nvim.git` into a fresh `~/.config/nvim` and tried a `git commit -v --amend` in one of my git repos; the colors for the commit changes are not there (the commit message itself IS correctly highlighted, though).

I then commented `init.lua:296` to look like this, and the commit changes are now properly highlited:
```diff
diff --git a/init.lua b/init.lua
index 7289032..04aa472 100644
--- a/init.lua
+++ b/init.lua
@@ -293,7 +293,7 @@ require('nvim-treesitter.configs').setup {
   -- Autoinstall languages that are not installed. Defaults to false (but you can change for yourself!)
   auto_install = false,
 
-  highlight = { enable = true },
+  -- highlight = { enable = true },
   indent = { enable = true, disable = { 'python' } },
   incremental_selection = {
     enable = true,
```

In case it helps: this is with `NVIM v0.8.3` running on `zsh 5.9`, on `arm-apple-darwin22.2.0`.

**Comment by kang8** [comment]
I created a brand new VPS and tried to reproduce the bug, but still couldn't reproduce it.

this is with `NVIM v0.8.3` running on `bash 5.1.16`, on `archlinux`.

---

Perhaps you can try cleaning the [runtime path](https://neovim.io/doc/user/options.html#'rtp') cache: 

```sh
rm -rf ~/.cache/nvim
rm -rf ~/.local/share/nvim
rm -rf ~/.local/state/nvim
```

> :warning: Please note that the above actions will delete cache files, installed plugins by lazy.nvim, and installed language server by mason. Please use with caution.

**Comment by gonzus** [comment]
Ok, some weird interactions here. I cleaned up those three directories, and let neovim rebuild everything. One of the things it rebuilds are the TreeSitter parsers living in `~/.local/share/nvim/lazy/nvim-treesitter/parser/`. The weird things are:
1. Sometimes these parsers are built for `x86_64`, although I am running under `arm`. I could not determine exactly how or when this happens.
2. If the `gitcommit.so` parser does not exist in this directory, and I try to amend a commit, syntax highlighting works 100% fine.
3. If the `gitcommit.so` parser exists (and is the correct `arm` type of shared object), and I try to amend a commit, syntax highlighting stops working, and I only see the colors for the commit message.

**Comment by kang8** [comment]
I know why now.

`gitcommit` can only parse git commit subject, message and generated, If you want to highlight the diff, you need to execute `:TSInstall diff` to install diff in tree-sitter. see [source](https://github.com/gbprod/tree-sitter-gitcommit/blob/f71b93f399c9c2b315825827c95466e7405ec622/README.md?plain=1#L38-L41)

**Comment by gonzus** [comment]
Hey @kang8 thanks, that at least explains part of the weirdness. So I added `diff` to the list of TS parsers, and now at least the `diff` lines are highlighted (`diff`, `index`, lines starting with `@@`); but the actual changed lines of code (and the surounding code context) are not. This is `C` code, for which I do have the TS parser installed, and when I open one of these `C` source files they do highlight properly.

I also installed the `git_rebase` parser, as mentioned in the code you pointed to, but that made no difference.

If I remove the `gitcommit` TS parser and immediately after I do a `git commit -v --amend`, two things happen:
1. The `diff` after the commit message comes up correctly highlighted, in full.
2. `nvim` notices I am missing the `gitcommit` parser and installs it for me, but it incorrectly compiles it for `x86_64` instead of `arm`. If I then remove the compiled parser and run `nvim` again (no args), the missing parser is installed again, this time compiled for the correct `arm` architecture.
